addons/point_of_sale/static/src/js/Screens/ProductScreen/ProductScreen.js

```diff
@@ -240,10 +258,15 @@ odoo.define('point_of_sale.ProductScreen', function(require) {
                     await this.env.pos._addProducts(foundProductIds);
                     // assume that the result is unique.
                     product = this.env.pos.db.get_product_by_id(foundProductIds[0]);
-                } else {
-                    return this._barcodeErrorAction(code);
                 }
             }
+            return product
+        }
+        async _barcodeProductAction(code) {
+            const product = await this._getProductByBarcode(code);
+            if (!product) {
+                return this._barcodeErrorAction(code);
+            }
```

Comment:
Consider the catch in the try-catch statement. There, it shows a popup and return undefined. Then in the product action, it will again show another error.

Reply:
@caburj I modified it.